packages/medusa/src/loaders/medusa-app.ts

```diff
@@ -48,18 +99,25 @@ export const loadMedusaApp = async (
   }
 
   container.register("remoteLink", asValue(medusaApp.link))
+  container.register(
+    ContainerRegistrationKeys.REMOTE_QUERY,
+    asValue(medusaApp.query)
+  )
 
-  const { query, modules } = medusaApp
-
-  // Medusa app load all non legacy modules, so we need to register them in the container since they are into their own container
-  // We might decide to do it elsewhere but for now I think it is fine
-  for (const [serviceKey, moduleService] of Object.entries(modules)) {
+  for (const [serviceKey, moduleService] of Object.entries(medusaApp.modules)) {
     container.register(
       ModulesDefinition[serviceKey].registrationName,
       asValue(moduleService)
     )
   }
-  container.register("remoteQuery", asValue(query))
+
+  // Register all unresolved modules as undefined to be present in the container with undefined value by defaul
+  // but still resolvable
+  for (const [, moduleDefinition] of Object.entries(ModulesDefinition)) {
+    if (!container.hasRegistration(moduleDefinition.registrationName)) {
```

Comment:
question: Can't we use Awilix to not fail on resolving non-existing dependencies? Or is this due to something else?

Reply:
yes we can set that on the main container to not throw on unregistered deps. would you prefer? It would also means that if someone register a service and try to inject it in some places, if we can't resolve i it wont throw, the server will start and it will throw when something will try to use it. Which can be problematic, what do you think?

for the modules we choose to do that, but not sure we want that for everything

Reply:
Hmm, yeah we probably don't want to do that for everything

Reply:
Fine to keep it as is